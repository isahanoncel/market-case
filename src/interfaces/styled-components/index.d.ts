// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft' | 'Check';

export type Colors =
  | 'primary'
  | 'gray'
  | 'secondary'
  | 'whiteBlue'
  | 'white'
  | 'background'
  | 'transparent'
  | 'innerBorder'
  | 'black100'
  | 'thinGray'
  | 'black500'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
