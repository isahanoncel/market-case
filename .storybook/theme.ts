import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
