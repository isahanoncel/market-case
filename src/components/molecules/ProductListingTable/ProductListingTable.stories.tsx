import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import ProductListingTable from '.';

export default {
  title: 'Molecules/ProductListingTable',
  component: ProductListingTable,
} as ComponentMeta<typeof ProductListingTable>;

export const Default: ComponentStory<typeof ProductListingTable> = (args) => (
  <ProductListingTable {...args} />
);
