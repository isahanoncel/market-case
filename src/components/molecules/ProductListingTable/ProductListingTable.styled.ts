import styled from 'styled-components';
import { device } from '../../../constants/styles/GlobalStyle';

export const ProductListingTableWrapper = styled.div`
  background-color: ${({ theme }) => theme.colors.white};
  box-shadow: 0px 4px 24px rgba(93, 62, 188, 0.04);
  border-radius: 2px;
  padding: 20px;
  margin-top: 16px;
  display: inline-flex;
  justify-content: space-around;
  flex-wrap: wrap;
  width: 608px;
  gap: 20px 24px;
  ${device.mobile} {
    width: 100%;
    display: block;
  }
`;

export const ProductListingTableItem = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 124px;
  height: 225px;

  ${device.mobile} {
    text-align: center;
    width: 100%;
  }
`;

export const ItemPrice = styled.div`
  color: ${({ theme }) => theme.colors.primary};
  font-weight: 700;
  font-size: 14px;
  margin-top: 8px;
  line-height: 20px;
`;

export const ItemImageWrapper = styled.div`
  padding: 16px;
  border: 1.17666px solid #f3f0fe;
  border-radius: 12px;
`;
