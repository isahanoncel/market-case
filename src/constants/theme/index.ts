import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
