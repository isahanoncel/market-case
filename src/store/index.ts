import { configureStore } from '@reduxjs/toolkit';
import { useDispatch, useSelector, TypedUseSelectorHook } from 'react-redux';
import products from './slices/products';
import basket from './slices/basket';

const store = configureStore({
  reducer: {
    products,
    basket,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export default store;
