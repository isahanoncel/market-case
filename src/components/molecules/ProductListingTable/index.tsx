import React, { FC, useMemo } from 'react';
import { IProduct } from '../../../interfaces/product';
import Button from '../../atoms/Button';
import Label from '../../atoms/Label';
import * as S from './ProductListingTable.styled';

interface IProductListingTable {
  items: IProduct[];
}

const ProductListingTable: FC<IProductListingTable> = ({ items }) => {
  const renderItems = useMemo(() => {
    return items.map((item) => {
      return (
        <S.ProductListingTableItem key={item.slug}>
          <S.ItemImageWrapper>
            <img
              src="https://i.picsum.photos/id/527/92/92.jpg?hmac=3exqIu160Nj6P4D-V4inAd0GOMHbsFdVaqDVJSweLoY"
              alt="market"
            />
          </S.ItemImageWrapper>
          <S.ItemPrice>â‚º {item.price}</S.ItemPrice>
          <Label size={14} weight={600}>
            {item.name}
          </Label>
          <Button fullWidth margin="8px 0 0 0">
            Add
          </Button>
        </S.ProductListingTableItem>
      );
    });
  }, [items]);

  return (
    <S.ProductListingTableWrapper>{renderItems}</S.ProductListingTableWrapper>
  );
};

export default ProductListingTable;
