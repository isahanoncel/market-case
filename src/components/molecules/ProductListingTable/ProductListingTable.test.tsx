import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import ProductListingTable from '.';
import theme from '../../../constants/theme';
import store from '../../../store';

test('ProductListingTable snapshot testing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <ProductListingTable items={[]} />
        </ThemeProvider>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
