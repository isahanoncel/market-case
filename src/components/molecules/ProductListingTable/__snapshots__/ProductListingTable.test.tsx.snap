// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListingTable snapshot testing 1`] = `
.c0 {
  background-color: #fff;
  box-shadow: 0px 4px 24px rgba(93,62,188,0.04);
  border-radius: 2px;
  padding: 20px;
  margin-top: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 608px;
  gap: 20px 24px;
}

@media (min-width:0px) and (max-width:768px) {
  .c0 {
    width: 100%;
    display: block;
  }
}

<div
  className="c0"
/>
`;
